application/models/artifact_builder.py

```diff
@@ -49,20 +40,50 @@ def _add_tags_to_neo(self, tags):
             self.neo.tags.connect(created_tag)
         self.neo.save()
 
-    def _overwrite_tags_in_neo(self, tags):
-        self.neo.tags.disconnect_all()
-        self._add_tags_to_neo(tags)
+    def _add_user_tags_to_neo(self, user_tags):
+        for tag in user_tags:
+            print(self.neo)
+            print(self.neo.user_tags)
+            created_tag = Tag.find_or_create_by(name=tag)
+            self.neo.user_tags.connect(created_tag)
+        self.neo.save()
+
+    def _overwrite_user_tags_in_neo(self, user_tags):
+        self.neo.user_tags.disconnect_all()
+        self._add_user_tags_to_neo(user_tags)
 
-    def update_with(self, override_tags=True, **properties):
-        """Update both models"""
-        neo_properties = self._make_properties_neo_compatible(**properties)
+    def _add_label_tags_to_neo(self, label_tags):
+        for tag in label_tags:
+            created_tag = Tag.find_or_create_by(name=tag)
+            self.neo.label_tags.connect(created_tag)
+        self.neo.save()
+
+    def _add_text_tags_to_neo(self, text_tags):
+        for tag in text_tags:
+            created_tag = Tag.find_or_create_by(name=tag)
+            self.neo.text_tags.connect(created_tag)
+        self.neo.save()
```

Comment:
These methods can easily be dried by giving them an extra parameter `tagType`

Reply:
done